core/src/com/unciv/logic/automation/city/ConstructionAutomation.kt

```diff
@@ -334,77 +315,57 @@ class ConstructionAutomation(val cityConstructions: CityConstructions) {
-            modifier *= 1.4f
-        modifier *= personality.scaledFocus(PersonalityValue.Science)
-        addChoice(relativeCostEffectiveness, scienceBuilding.name, modifier)
-    }
+        if (buildingStats.gold < 0 && civInfo.gold < 0) {
+            buildingStats.gold *= 2 // We have a gold problem and this isn't helping
+        }
 
-    private fun addGoldBuildingChoice() {
-        val goldBuilding = statBuildings.filter { it.isStatRelated(Stat.Gold) }
-            .filterBuildable()
-            .minByOrNull { it.cost } ?: return
-        var modifier = if (civInfo.stats.statsForNextTurn.gold < 0) 3f else 1.2f
-        modifier *= personality.scaledFocus(PersonalityValue.Gold)
-        addChoice(relativeCostEffectiveness, goldBuilding.name, modifier)
-    }
+        if (civInfo.getHappiness() < 0)
+            buildingStats.happiness *= 2
 
-    private fun addProductionBuildingChoice() {
-        val productionBuilding = statBuildings
-            .filter { it.isStatRelated(Stat.Production) }
-            .filterBuildable()
-            .minByOrNull { it.cost } ?: return
-        val modifier = personality.scaledFocus(PersonalityValue.Production)
-        addChoice(relativeCostEffectiveness, productionBuilding.name, 1.5f * modifier)
-    }
+        if (city.cityStats.currentCityStats.culture < 1) {
+            buildingStats.culture *= 2 // We need to start growing borders
+        }
+        else if (city.tiles.size < 12 && city.population.population < 5) {
+            buildingStats.culture *= 2
+        }
 
-    private fun addFoodBuildingChoice() {
-        val conditionalState = StateForConditionals(civInfo, city)
-        val foodBuilding = nonWonders
-            .filter {
-                (it.isStatRelated(Stat.Food)
-                    || it.hasUnique(UniqueType.CarryOverFood, conditionalState)
-                ) && Automation.allowAutomatedConstruction(civInfo, city, it)
-            }.filterBuildable().minByOrNull { it.cost } ?: return
-        var modifier = 1f
-        if (city.population.population < 5) modifier = 1.3f
-        modifier *= personality.scaledFocus(PersonalityValue.Food)
-        addChoice(relativeCostEffectiveness, foodBuilding.name, modifier)
+        for (stat in Stat.values()) {
+            if (
+                stat == Stat.Culture && civInfo.wantsToFocusOn(Victory.Focus.Culture) ||
+                stat == Stat.Production && civInfo.wantsToFocusOn(Victory.Focus.Production) ||
+                stat == Stat.Science && civInfo.wantsToFocusOn(Victory.Focus.Science) ||
+                stat == Stat.Faith && civInfo.wantsToFocusOn(Victory.Focus.Faith) ||
+                stat == Stat.Gold && civInfo.wantsToFocusOn(Victory.Focus.Gold)
+                )
+                buildingStats[stat] *= 2f
+
+            buildingStats[stat] *= personality.scaledFocus(PersonalityValue[stat])
+        }
 
+        building.happiness /= 3 // Avoid overshooting that may happen from the following rank
```

Comment:
Happiness is usually VERY important, there's a reason that we mark it as 3 times the regular stat. I wouldn't consider this overshooting, if it is - we should be changing the rankStatsValue, because that overshoot would apply everywhere (specialists, tiles to work, improvements, etc)

Reply:
> Happiness is usually VERY important, there's a reason that we mark it as 3 times the regular stat

It's not uncommon for higher level AIs to be left with something like +20 happiness thanks to the many multipliers they get. There, happiness genuinely doesn't matter nearly as much

> because that overshoot would apply everywhere (specialists, tiles to work, improvements, etc)

Also no. It actually is only used in places of placing new cities and building improvements. Tiles to work/specialists to work actually has their own way of ranking that's separate. Tbh, I'm only using rankStatsValue for consistency and to future proof for tourism, but even here it's kinda not even that necessary to take into account. I could be willing to edit rankStatsValue (it basically only comes up for placing cities... I think... due to very few improvements, even in mods, actually affecting happiness)

Reply:
If so, we should add that as a precondition - if civ happiness is larger than some metric (say, max(15, numOfCities)), then divide by 3; if larger than half that, divide by 2; else leave as-is

Reply:
Done in my last commit... Eh... On second thought, not sure I like it. As I said, tiles to work/specialists to work uses it's own value system, and I based most of the value system here off of it, only actually using the prebuilt Ranking for any broad strokes things or for specifically viewing gold less than other stats. As such, line 331 already tries to take this into account and I'm now overshooting again (x6 value instead of the intended x3 value). For me the question would be "do we always care about happiness the same way", to which the answer in Unciv is yes, but in Civ 5 is no (Civ 5 has a distinction between local happiness and global). So if we made changes to be 100% consistent with Civ 5, it actually is better to consider the happiness city by city, rather than with the global modifier

Fun fact, btw: tiles to work also overshoots happiness by accident, intending to value it at x2 when unhappy and x1 otherwise, but having the check twice by accident, leading to it actually being x4. It's actually not the only accidental mistake I noticed while glancing at it